src/test/java/api/loans/CheckOutByBarcodeScheduledNoticesTests.java

```diff
@@ -0,0 +1,232 @@
+    Period beforeRecurringPeriod = Period.hours(6);
+
+    UUID uponAtTemplateId = UUID.randomUUID();
+
+    UUID afterTemplateId = UUID.randomUUID();
+    Period afterPeriod = Period.days(3);
+    Period afterRecurringPeriod = Period.hours(4);
+
+    JsonObject beforeDueDateNoticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(beforeTemplateId)
+      .withDueDateEvent()
+      .withBeforeTiming(beforePeriod)
+      .recurring(beforeRecurringPeriod)
+      .sendInRealTime(true)
+      .create();
+    JsonObject uponAtDueDateNoticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(uponAtTemplateId)
+      .withDueDateEvent()
+      .withUponAtTiming()
+      .sendInRealTime(false)
+      .create();
+    JsonObject afterDueDateNoticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(afterTemplateId)
+      .withDueDateEvent()
+      .withAfterTiming(afterPeriod)
+      .recurring(afterRecurringPeriod)
+      .sendInRealTime(true)
+      .create();
+    NoticePolicyBuilder noticePolicy = new NoticePolicyBuilder()
+      .withName("Policy with due date notices")
+      .withLoanNotices(Arrays.asList(
+        beforeDueDateNoticeConfiguration,
+        uponAtDueDateNoticeConfiguration,
+        afterDueDateNoticeConfiguration));
+    useLoanPolicyAsFallback(
+      loanPoliciesFixture.canCirculateRolling().getId(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.create(noticePolicy).getId());
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource steve = usersFixture.steve();
+
+    final DateTime loanDate =
+      new DateTime(2018, 3, 18, 11, 43, 54, DateTimeZone.UTC);
+
+    final IndividualResource loan = loansFixture.checkOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(smallAngryPlanet)
+        .to(steve)
+        .on(loanDate)
+        .at(UUID.randomUUID()));
+    DateTime dueDate = new DateTime(loan.getJson().getString("dueDate"));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticeClient::getAll, Matchers.hasSize(3));
+
+    List<JsonObject> scheduledNotices = scheduledNoticeClient.getAll();
+    MatcherAssert.assertThat(scheduledNotices,
+      hasItems(
+        hasScheduledLoanNoticeProperties(
+          loan.getId(), dueDate.minus(beforePeriod.timePeriod()),
+          BEFORE_TIMING, beforeTemplateId,
+          beforeRecurringPeriod, true),
+        hasScheduledLoanNoticeProperties(
+          loan.getId(), dueDate,
+          UPON_AT_TIMING, uponAtTemplateId,
+          null, false),
+        hasScheduledLoanNoticeProperties(
+          loan.getId(), dueDate.plus(afterPeriod.timePeriod()),
+          AFTER_TIMING, afterTemplateId,
+          afterRecurringPeriod, true)
+      ));
+  }
+
+  @Test
+  public void multipleBeforeDueDateNoticesShouldBeScheduledOnCheckout()
```

Comment:
What is the difference between this test and the one above?

Reply:
It checks the ability to create several notices of the same type (like "Before" notice).

Reply:
I can't say this test checks something important.

Reply:
Maybe renaming this to `checkOutSchedulesDifferentBeforeDueDateNotices` could help make that clearer?

Reply:
Thank you for helping me understand the difference

Reply:
Thanks for the suggestion, renamed method.